core/domain/skill_services_test.py

```diff
@@ -775,6 +775,20 @@ def test_delete_skill(self):
             skill_services.get_skill_summary_by_id(
                 self.SKILL_ID, strict=False), None)
 
+    def test_delete_skill_marked_deleted(self):
+        skill_models.SkillModel.delete_multi(
+            [self.SKILL_ID], self.USER_ID, '', force_deletion=False)
+        skill_model = skill_models.SkillModel.get_by_id(self.SKILL_ID)
+        self.assertTrue(skill_model.deleted)
+
+        skill_services.delete_skill(
+            self.USER_ID, self.SKILL_ID, force_deletion=True)
+        skill_model = skill_models.SkillModel.get_by_id(self.SKILL_ID)
+        self.assertEqual(skill_model, None)
+        self.assertEqual(
+            skill_services.get_skill_summary_by_id(
+                self.SKILL_ID, strict=False), None)
```

Comment:
What about the case when the summary model is already deleted, should we test that?

Reply:
Or the case when everything is done except for `suggestion_services.auto_reject_question_suggestions_for_skill_id`.

Reply:
We should very carefully test all the possibilities.

Reply:
Done.